dask_cloudprovider/providers/azure/azureml.py

```diff
@@ -715,9 +717,9 @@ def update():
 
     def close_when_disconnect(self):
         if (
-            self.run.get_status() == "Canceled"
-            or self.run.get_status() == "Completed"
-            or self.run.get_status() == "Failed"
+            self.run.status == "Canceled"
```

Comment:
again here, i'd consider something like the below to get the latest status once as this call could be disjoint in time from the previous refresh

```
status = run.get_status()
if status == A or status == B....: